src/delegation_backend/aws_keyspaces.go

```diff
@@ -211,21 +197,66 @@ func calculateShard(submittedAt time.Time) int {
 	return (3600*hour + 60*minute + second) / 144
 }
 
-func (kc *KeyspaceContext) tryInsertSubmission(submission *Submission, includeRawBlock bool) error {
-	query := "INSERT INTO " + kc.Keyspace + ".submissions (submitted_at_date, shard, submitted_at, submitter, remote_addr, peer_id, snark_work, block_hash, created_at, graphql_control_port, built_with_commit_sha"
-	values := []interface{}{submission.SubmittedAtDate, calculateShard(submission.SubmittedAt), submission.SubmittedAt, submission.Submitter, submission.RemoteAddr, submission.PeerId, submission.SnarkWork, submission.BlockHash, submission.CreatedAt, submission.GraphqlControlPort, submission.BuiltWithCommitSha}
-	if includeRawBlock {
-		query += ", raw_block) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
-		values = append(values, submission.RawBlock)
-	} else {
-		query += ") VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)"
+// Estimate the size of the raw block in bytes.
+// In Go, len() returns the number of bytes in a slice, which should suffice for a rough estimation.
+func calculateBlockSize(rawBlock []byte) int {
+	return len(rawBlock)
+}
+
+// Insert a submission into the Keyspaces database
+func (kc *KeyspaceContext) insertSubmission(submission *Submission) error {
+	return ExponentialBackoff(func() error {
+		if submission.RawBlock != nil && calculateBlockSize(submission.RawBlock) > MAX_BLOCK_SIZE {
+			kc.Log.Infof("KeyspaceSave: Block too large (%d bytes), inserting without raw_block", calculateBlockSize(submission.RawBlock))
+			if err := kc.insertSubmissionWithoutRawBlock(submission); err != nil {
+				return err
+			}
+		} else {
+			if err := kc.insertSubmissionWithRawBlock(submission); err != nil {
+				return err
+			}
+
+		}
```

Comment:
Although I think this should work, it's still confusing, because in case the `RawBlock` is `nil`, we insert into Cassandra with raw block. Besides, when can be the raw block missing from the submission? Shouldn't we raise an error if it's absent?